apps/minifront/src/state/staking.ts

```diff
@@ -162,26 +169,67 @@ export const createStakingSlice = (): SliceCreator<StakingSlice> => (set, get) =
     set(state => {
       state.staking.delegationsByAccount.set(addressIndex.account, []);
       state.staking.votingPowerByValidatorInfo = {};
+      state.staking.loading = true;
     });
 
-    for await (const delegation of getDelegationsForAccount(addressIndex)) {
-      const delegations = get().staking.delegationsByAccount.get(addressIndex.account) ?? [];
+    let delegationsToFlush: ValueView[] = [];
+    let allDelegationsLoaded = false;
 
-      const sortedDelegations = [...delegations, delegation].sort(byBalanceAndVotingPower);
+    /**
+     * Per the RPC call, we get delegations in a stream, one-by-one. If we push
+     * them to state as we receive them, React has to re-render super
+     * frequently. Rendering happens synchronously, which means that the `for`
+     * loop below has to wait until rendering is done before moving on to the
+     * next delegation. Thus, the staking page loads super slowly if we render
+     * delegations as soon as we receive them.
+     *
+     * To resolve this performance issue, we instead queue up a number of
+     * delegations (up to `DELEGATIONS_FLUSH_TO_STATE_INTERVAL`), and then flush
+     * them to state in batches.
+     */
+    const flushToState = (ignoreInterval = false) => {
```

Comment:
I wonder if there is a more generic throttle helper function that could be made here 🤔